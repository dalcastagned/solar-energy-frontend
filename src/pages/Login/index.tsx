import { useState } from 'react';

import EmailIcon from '@mui/icons-material/Email';
import LockIcon from '@mui/icons-material/Lock';
import { Fade, InputAdornment } from '@mui/material';
import { useFormik } from 'formik';
import { useHistory } from 'react-router-dom';
import * as Yup from 'yup';

import { useAuth } from '../../hooks/auth';
import { LoginCredentials } from '../../models/user';
import { ErrorResponse, handleErrorResponse } from '../../utils/error-response';

import * as S from './styles';

const validationSchema = Yup.object({
  email: Yup.string()
    .email('Entre com um email válido')
    .required('Email é obrigatório'),
  password: Yup.string()
    .min(6, 'Senha precisa ter no mínimo 6 caracteres')
    .max(60, 'Máximo de 60 caracteres')
    .required('Senha é obrigatória'),
});

const Login = (): JSX.Element => {
  const { signIn } = useAuth();
  const [errorMessage, setErrorMessage] = useState('');
  const [loading, setLoading] = useState(false);
  const history = useHistory();

  const handleLogin = async (credentials: LoginCredentials): Promise<void> => {
    try {
      setLoading(true);
      await signIn(credentials);
    } catch (err) {
      const errorMessageToValidate = (
        err as ErrorResponse
      ).response?.data?.toString();

      if (errorMessageToValidate?.includes('Password expired')) {
        history.push(`/reset-password`, { email: credentials.email });
      }
      setErrorMessage(handleErrorResponse(err));
      setLoading(false);
    }
  };

  const onCloseAlert = (): void => {
    setErrorMessage('');
  };

  const formik = useFormik({
    initialValues: {
      email: '',
      password: '',
    },
    validationSchema,
    onSubmit: values => {
      handleLogin(values);
    },
  });

  return (
    <S.Wrapper>
      <S.ContainerImage />
      <Fade in timeout={1000}>
        <S.ContainerLogin>
          <S.ContainerInfo>
            <img src="/img/logo.png" alt="Logo" />
            <h1>Seja bem vindo</h1>
          </S.ContainerInfo>
          <S.Form onSubmit={formik.handleSubmit}>
            {errorMessage && (
              <S.AlertStyled severity="error" onClose={onCloseAlert}>
                {errorMessage}
              </S.AlertStyled>
            )}
            <S.TextFieldStyled
              id="email"
              name="email"
              label="Email"
              InputProps={{
                placeholder: 'exemplo@exemplo.com',
                startAdornment: (
                  <InputAdornment position="start">
                    <EmailIcon />
                  </InputAdornment>
                ),
              }}
              value={formik.values.email}
              onChange={formik.handleChange}
              error={formik.touched.email && Boolean(formik.errors.email)}
              helperText={formik.touched.email && formik.errors.email}
            />
            <S.TextFieldStyled
              id="password"
              name="password"
              label="Senha"
              type="password"
              InputProps={{
                placeholder: 'Digite sua senha',
                startAdornment: (
                  <InputAdornment position="start">
                    <LockIcon />
                  </InputAdornment>
                ),
              }}
              value={formik.values.password}
              onChange={formik.handleChange}
              error={formik.touched.password && Boolean(formik.errors.password)}
              helperText={formik.touched.password && formik.errors.password}
            />
            <S.ButtonStyled
              color="primary"
              variant="contained"
              fullWidth
              type="submit"
              disabled={loading}
            >
              Entrar
            </S.ButtonStyled>
          </S.Form>
          <S.LinkStyled href="/signup">
            Ainda não possui uma conta? Cadastre-se!
          </S.LinkStyled>
        </S.ContainerLogin>
      </Fade>
    </S.Wrapper>
  );
};

export default Login;
