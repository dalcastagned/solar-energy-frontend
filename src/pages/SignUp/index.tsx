import { useState } from 'react';

import EmailIcon from '@mui/icons-material/Email';
import LockIcon from '@mui/icons-material/Lock';
import { Fade, InputAdornment } from '@mui/material';
import { useFormik } from 'formik';
import { useHistory, useLocation } from 'react-router-dom';
import * as Yup from 'yup';

import { useAuth } from '../../hooks/auth';
import { handleErrorResponse } from '../../utils/error-response';

import * as S from './styles';

const validationSchema = Yup.object().shape({
  email: Yup.string()
    .email('Entre com um email válido')
    .required('Email é obrigatório'),
  password: Yup.string()
    .min(6, 'Senha precisa ter no mínimo 6 caracteres')
    .required('A senha é obrigatória.')
    .max(60, 'Máximo de 60 caracteres'),
  confirmPassword: Yup.string()
    .min(6, 'Senha precisa ter no mínimo 6 caracteres')
    .required('A confirmação de senha é obrigatória.')
    .oneOf([Yup.ref('password'), null], 'Senhas não correspondem')
    .max(60, 'Máximo de 60 caracteres'),
});

const SignUp = (): JSX.Element => {
  const { signUp } = useAuth();
  const [errorMessage, setErrorMessage] = useState('');
  const [loading, setLoading] = useState(false);
  const history = useHistory();
  const location = useLocation();

  const handleSignUp = async (
    email: string,
    password: string,
  ): Promise<void> => {
    try {
      setLoading(true);
      await signUp({
        email,
        password,
      });
      history.push('/');
    } catch (err) {
      setErrorMessage(handleErrorResponse(err, location.pathname));
      setLoading(false);
    }
  };

  const onCloseAlert = (): void => {
    setErrorMessage('');
  };

  const formik = useFormik({
    initialValues: {
      email: '',
      password: '',
      confirmPassword: '',
    },
    validationSchema,
    onSubmit: values => {
      handleSignUp(values.email, values.password);
    },
  });

  return (
    <S.Wrapper>
      <S.ContainerImage />
      <Fade in timeout={1000}>
        <S.ContainerSignUp>
          <S.ContainerInfo>
            <img src="/img/logo.png" alt="Logo" />
            <h1>Seja bem-vindo</h1>
          </S.ContainerInfo>
          <S.Form onSubmit={formik.handleSubmit}>
            {errorMessage && (
              <S.AlertStyled severity="error" onClose={onCloseAlert}>
                {errorMessage}
              </S.AlertStyled>
            )}
            <S.TextFieldStyled
              id="email"
              name="email"
              label="Email"
              InputProps={{
                placeholder: 'exemplo@exemplo.com',
                startAdornment: (
                  <InputAdornment position="start">
                    <EmailIcon />
                  </InputAdornment>
                ),
              }}
              value={formik.values.email}
              onChange={formik.handleChange}
              error={formik.touched.email && Boolean(formik.errors.email)}
              helperText={formik.touched.email && formik.errors.email}
            />
            <S.TextFieldStyled
              id="password"
              name="password"
              label="Senha"
              type="password"
              InputProps={{
                placeholder: 'Digite sua senha',
                startAdornment: (
                  <InputAdornment position="start">
                    <LockIcon />
                  </InputAdornment>
                ),
              }}
              value={formik.values.password}
              onChange={formik.handleChange}
              error={formik.touched.password && Boolean(formik.errors.password)}
              helperText={formik.touched.password && formik.errors.password}
            />
            <S.TextFieldStyled
              id="confirmPassword"
              name="confirmPassword"
              label="Confirmar senha"
              type="password"
              InputProps={{
                placeholder: 'Confirme sua senha',
                startAdornment: (
                  <InputAdornment position="start">
                    <LockIcon />
                  </InputAdornment>
                ),
              }}
              value={formik.values.confirmPassword}
              onChange={formik.handleChange}
              error={
                formik.touched.confirmPassword &&
                Boolean(formik.errors.confirmPassword)
              }
              helperText={
                formik.touched.confirmPassword && formik.errors.confirmPassword
              }
            />
            <S.ButtonStyled
              color="primary"
              variant="contained"
              fullWidth
              type="submit"
              disabled={loading}
            >
              Cadastrar
            </S.ButtonStyled>
          </S.Form>
          <S.LinkStyled href="/">
            Já possui uma conta? Faça o login!
          </S.LinkStyled>
        </S.ContainerSignUp>
      </Fade>
    </S.Wrapper>
  );
};

export default SignUp;
