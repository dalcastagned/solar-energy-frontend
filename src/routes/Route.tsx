import { ComponentType } from 'react';

import {
  Redirect,
  Route as ReactDOMRoute,
  RouteComponentProps,
  RouteProps as ReactDOMRouteProps,
} from 'react-router-dom';
import { whereRedirect } from 'utils/tests/where-redirect';

import { useAuth } from '../hooks/auth';

export type RouteProps = {
  isPrivate?: boolean;
  Component: ComponentType<RouteComponentProps>;
  roles?: string[];
} & ReactDOMRouteProps;

const Route = ({
  isPrivate = false,
  Component,
  ...props
}: RouteProps): JSX.Element => {
  const { user } = useAuth();

  const render = (params: RouteComponentProps): React.ReactNode => {
    const isAuthenticated = isPrivate === !!user;

    if (!isAuthenticated) {
      return (
        <Redirect
          to={{
            pathname: whereRedirect(isPrivate),
            state: { from: params.location },
          }}
        />
      );
    }

    return <Component {...params} />;
  };

  return <ReactDOMRoute {...props} render={render} />;
};

export default Route;
