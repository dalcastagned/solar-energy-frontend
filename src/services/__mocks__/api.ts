import axios from 'axios';

export type Error = {
  status?: number;
  data?: Record<string, unknown>;
  message?: string;
};

export type HttpResponse<DataType> = {
  data?: DataType;
  error?: Error;
};

const api = jest.genMockFromModule('axios') as jest.MockedFunction<
  typeof axios
>;

api.create = jest.fn(() => api);

export default api;
