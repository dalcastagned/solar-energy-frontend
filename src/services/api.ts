import axios, { AxiosError, AxiosResponse } from 'axios';

// import { cleanStorage } from '../hooks/auth';

export type Error = {
  status?: number;
  data?: Record<string, unknown>;
  message?: string;
};

export type HttpResponse<DataType> = {
  data?: DataType;
  error?: Error;
};

export type ErrorResponse = {
  response: Error;
  code: string;
};

const responseSuccessHandler = (response: AxiosResponse): AxiosResponse =>
  response;

const forbidden = [401, 403];

const responseErrorHandler = (error: AxiosError): Promise<AxiosError> => {
  const isUnauthorized =
    error.response && forbidden.includes(error.response.status);
  const routers = ['/', '/reset-password'].includes(window.location.pathname);
  if (!routers && isUnauthorized) {
    // cleanStorage();
    window.location.href = '/';
  }

  return Promise.reject(error);
};

const api = axios.create({ baseURL: 'http://localhost:7140/api' });

api.interceptors.response.use(
  response => responseSuccessHandler(response),
  error => responseErrorHandler(error),
);

export default api;
