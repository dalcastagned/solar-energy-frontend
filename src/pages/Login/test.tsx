import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Login from '.';
import { AuthProvider } from '../../hooks/auth';
import Route from '../../routes/Route';
import api from '../../services/api';
import { renderWithRoute } from '../../utils/tests/helpers';

describe('<Login />', () => {
  it('should render the main', () => {
    const { container } = renderWithRoute(<Login />);
    expect(screen.getByText('Entrar')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the validation messages when login without username and password', async () => {
    renderWithRoute(<Login />);

    const loginButton = screen.getByText('Entrar');

    userEvent.click(loginButton);

    await waitFor(() => {
      expect(screen.getByText('Email é obrigatório')).toBeInTheDocument();
      expect(screen.getByText('Senha é obrigatória')).toBeInTheDocument();
    });
  });

  it('must render validation messages when password contain more than 60 characters', async () => {
    renderWithRoute(<Login />);

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, Array(61).fill('a').join(''));

    const loginButton = screen.getByText('Entrar');

    userEvent.click(loginButton);

    await waitFor(() => {
      expect(screen.getByText('Máximo de 60 caracteres')).toBeInTheDocument();
    });
  });

  it('should render alert when user is not authorized', async () => {
    jest.spyOn(api, 'post').mockImplementation(async () => {
      const error = { response: { status: 401 } };
      throw error;
    });

    renderWithRoute(
      <AuthProvider>
        <Login />
      </AuthProvider>,
    );

    const userInput = screen.getByLabelText('Email');
    userEvent.type(userInput, 'usuario@usuario.com');

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, '123456');

    const loginButton = screen.getByText('Entrar');

    userEvent.click(loginButton);

    await waitFor(() => {
      expect(
        screen.getByText('Usuário ou senha inválidos.'),
      ).toBeInTheDocument();
    });
  });

  it('should render alert when Password Expired', async () => {
    jest.spyOn(api, 'post').mockRejectedValue({
      response: { status: 401, data: 'Password expired' },
    });

    const resetPage = (): JSX.Element => <h1>Alterar senha</h1>;

    renderWithRoute(
      <AuthProvider>
        <Route Component={Login} />
        <Route Component={resetPage} path="/reset-password" />
      </AuthProvider>,
    );

    const userInput = screen.getByLabelText('Email');
    userEvent.type(userInput, 'usuario@usuario.com');

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, '123456');

    const loginButton = screen.getByText('Entrar');

    userEvent.click(loginButton);

    await waitFor(() => {
      expect(screen.getByText('Alterar senha')).toBeInTheDocument();
    });
  });

  it('should render alert when server is down', async () => {
    jest.spyOn(api, 'post').mockImplementation(async () => {
      const error = { code: 'ECONNABORTED' };
      throw error;
    });

    renderWithRoute(
      <AuthProvider>
        <Login />
      </AuthProvider>,
    );

    const userInput = screen.getByLabelText('Email');
    userEvent.type(userInput, 'usuario@usuario.com');

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, '123456');

    const loginButton = screen.getByRole('button', { name: 'Entrar' });

    userEvent.click(loginButton);

    await waitFor(async () => {
      expect(
        screen.getByText(
          'O servidor parece estar indisponível. Verifique sua conexão.',
        ),
      ).toBeInTheDocument();
    });
  });

  it('should render alert with server error message', async () => {
    jest.spyOn(api, 'post').mockImplementation(async () => {
      const error = { response: { status: 500 } };
      throw error;
    });

    renderWithRoute(
      <AuthProvider>
        <Login />
      </AuthProvider>,
    );

    const userInput = screen.getByLabelText('Email');
    userEvent.type(userInput, 'usuario@usuario.com');

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, '123456');

    const loginButton = screen.getByText('Entrar');

    userEvent.click(loginButton);

    await waitFor(() => {
      expect(
        screen.getByText('Ocorreu um erro no servidor.'),
      ).toBeInTheDocument();
    });
  });

  it('should close alert when clicked on closeButton', async () => {
    jest.spyOn(api, 'post').mockImplementation(async () => {
      const error = { response: undefined };
      throw error;
    });

    renderWithRoute(
      <AuthProvider>
        <Login />
      </AuthProvider>,
    );

    const userInput = screen.getByLabelText('Email');
    userEvent.type(userInput, 'usuario@usuario.com');

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, '123456');

    const loginButton = screen.getByText('Entrar');

    userEvent.click(loginButton);

    await waitFor(() => {
      const closeAlert = screen.getByLabelText('Close');
      userEvent.click(closeAlert);

      expect(screen.queryByRole('alert')).not.toBeInTheDocument();
    });
  });
});
