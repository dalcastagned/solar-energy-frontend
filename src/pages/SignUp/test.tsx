import { act, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import SignUp from '.';
import Login from '.';
import { AuthProvider } from '../../hooks/auth';
import Route from '../../routes/Route';
import api from '../../services/api';
import { renderWithRoute } from '../../utils/tests/helpers';

describe('<SignUp />', () => {
  it('should render the main', () => {
    const { container } = renderWithRoute(<SignUp />);

    expect(screen.getByText('Cadastrar')).toBeInTheDocument();
    expect(screen.getByLabelText('Email')).toBeInTheDocument();
    expect(screen.getByLabelText('Senha')).toBeInTheDocument();
    expect(screen.getByLabelText('Confirmar senha')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the validation messages when sign up without email, passord and confirmPassword', async () => {
    renderWithRoute(<SignUp />);

    const signUpButton = screen.getByRole('button', {
      name: 'Cadastrar',
    });

    userEvent.click(signUpButton);

    await waitFor(() => {
      expect(screen.getByText('Email é obrigatório')).toBeInTheDocument();
      expect(screen.getByText('A senha é obrigatória.')).toBeInTheDocument();
      expect(
        screen.getByText('A confirmação de senha é obrigatória.'),
      ).toBeInTheDocument();
    });
  });

  it('must render validation messages when passord and confirmPassword contain more than 60 characters', async () => {
    renderWithRoute(<Login />);

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, Array(61).fill('a').join(''));

    const confirmPasswordInput = screen.getByLabelText('Confirmar senha');
    userEvent.type(confirmPasswordInput, Array(61).fill('a').join(''));

    const signUpButton = screen.getByRole('button', {
      name: 'Cadastrar',
    });

    userEvent.click(signUpButton);

    await waitFor(() => {
      const [passwordValidation, confirmPasswordValidation] =
        screen.getAllByText('Máximo de 60 caracteres');
      expect(passwordValidation).toBeInTheDocument();
      expect(confirmPasswordValidation).toBeInTheDocument();
    });
  });

  it('should render alert when user already exist', async () => {
    jest.spyOn(api, 'post').mockRejectedValue({ response: { status: 400 } });

    const history = createMemoryHistory();
    history.push(`/signup`);

    renderWithRoute(
      <AuthProvider>
        <Router history={history}>
          <SignUp />
        </Router>
      </AuthProvider>,
    );

    const emailInput = screen.getByLabelText('Email');
    userEvent.type(emailInput, 'usuario@usuario.com');

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, 'senha1');

    const confirmPasswordInput = screen.getByLabelText('Confirmar senha');
    userEvent.type(confirmPasswordInput, 'senha1');

    const signUpButton = screen.getByRole('button', {
      name: 'Cadastrar',
    });

    userEvent.click(signUpButton);

    await waitFor(() => {
      expect(screen.getByText('Usuário já cadastrado.')).toBeInTheDocument();
    });
  });

  it('should render alert when server is down', async () => {
    jest.spyOn(api, 'post').mockRejectedValue({ code: 'ECONNABORTED' });

    const history = createMemoryHistory();
    history.push(`/signup`);

    renderWithRoute(
      <AuthProvider>
        <Router history={history}>
          <SignUp />
        </Router>
      </AuthProvider>,
    );

    const emailInput = screen.getByLabelText('Email');
    userEvent.type(emailInput, 'usuario@usuario.com');

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, 'senha1');

    const confirmPasswordInput = screen.getByLabelText('Confirmar senha');
    userEvent.type(confirmPasswordInput, 'senha1');

    const signUpButton = screen.getByRole('button', {
      name: 'Cadastrar',
    });

    userEvent.click(signUpButton);

    await waitFor(async () => {
      expect(
        screen.getByText(
          'O servidor parece estar indisponível. Verifique sua conexão.',
        ),
      ).toBeInTheDocument();
    });
  });

  it('should render alert with server error message', async () => {
    jest.spyOn(api, 'post').mockRejectedValue({ response: { status: 500 } });

    const history = createMemoryHistory();
    history.push(`/signup`);

    renderWithRoute(
      <AuthProvider>
        <Router history={history}>
          <SignUp />
        </Router>
      </AuthProvider>,
    );

    const emailInput = screen.getByLabelText('Email');
    userEvent.type(emailInput, 'usuario@usuario.com');

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, 'senha1');

    const confirmPasswordInput = screen.getByLabelText('Confirmar senha');
    userEvent.type(confirmPasswordInput, 'senha1');

    const signUpButton = screen.getByRole('button', {
      name: 'Cadastrar',
    });

    userEvent.click(signUpButton);

    await waitFor(() => {
      expect(
        screen.getByText('Ocorreu um erro no servidor.'),
      ).toBeInTheDocument();
    });
  });

  it('should close alert when clicked on closeButton', async () => {
    jest.spyOn(api, 'post').mockRejectedValue({ response: undefined });

    const history = createMemoryHistory();
    history.push(`/signup`);

    renderWithRoute(
      <AuthProvider>
        <Router history={history}>
          <SignUp />
        </Router>
      </AuthProvider>,
    );

    const emailInput = screen.getByLabelText('Email');
    userEvent.type(emailInput, 'usuario@usuario.com');

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, 'senha1');

    const confirmPasswordInput = screen.getByLabelText('Confirmar senha');
    userEvent.type(confirmPasswordInput, 'senha1');

    const signUpButton = screen.getByRole('button', {
      name: 'Cadastrar',
    });

    userEvent.click(signUpButton);

    await waitFor(() => {
      const closeAlert = screen.getByLabelText('Close');
      userEvent.click(closeAlert);

      expect(screen.queryByRole('alert')).not.toBeInTheDocument();
    });
  });

  it('should render alert when password confirmation is different from password', async () => {

    renderWithRoute(
      <AuthProvider>
        <SignUp />
      </AuthProvider>,
    );

    const emailInput = screen.getByLabelText('Email');
    userEvent.type(emailInput, 'usuario@usuario.com');

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, 'senha1');

    const confirmPasswordInput = screen.getByLabelText('Confirmar senha');
    userEvent.type(confirmPasswordInput, 'senha2');

    const signUpButton = screen.getByRole('button', {
      name: 'Cadastrar',
    });

    userEvent.click(signUpButton);

    await waitFor(() => {
      expect(screen.getByText('Senhas não correspondem')).toBeInTheDocument();
    });
  });

  it('should go to login screen when sign up successful', async () => {
    jest.spyOn(api, 'post').mockImplementation(jest.fn());

    const history = createMemoryHistory();
    history.push(`/signup`);

    const loginPage = (): JSX.Element => <h1>Login</h1>;

    renderWithRoute(
      <AuthProvider>
        <Router history={history}>
          <SignUp />
          <Route Component={loginPage} path="/" />
        </Router>
      </AuthProvider>,
    );

    const emailInput = screen.getByLabelText('Email');
    userEvent.type(emailInput, 'usuario@usuario.com');

    const passwordInput = screen.getByLabelText('Senha');
    userEvent.type(passwordInput, 'senha1');

    const confirmPasswordInput = screen.getByLabelText('Confirmar senha');
    userEvent.type(confirmPasswordInput, 'senha1');

    const signUpButton = screen.getByRole('button', {
      name: 'Cadastrar',
    });

    await act(async () => {
      userEvent.click(signUpButton);

      await waitFor(() => {
        expect(screen.getByText('Login')).toBeInTheDocument();
      });
    });
  });
});
