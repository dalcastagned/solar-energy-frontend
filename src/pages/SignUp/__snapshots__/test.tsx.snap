// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> should render the main 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c1 {
  width: 50%;
  height: 100%;
  background: url('/img/login.png') center no-repeat;
  background-size: cover;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 50%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-animation: hUniol 1.5s;
  animation: hUniol 1.5s;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 50%;
}

.c3 img {
  max-width: 28rem;
}

.c3 h1 {
  font-size: 2.4rem;
  color: #374557;
  font-weight: 500;
  line-height: 3.2rem;
  text-align: center;
  padding: 4rem 0;
}

.c4 {
  width: 100%;
  padding: 0 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 3rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media screen and (max-width:980px) {
  .c1 {
    display: none;
  }
}

@media screen and (max-width:980px) {
  .c2 {
    width: 100%;
    background: linear-gradient( 135deg, rgba(255,255,255,0.9) -10%, rgba(255,255,255,0.9) 180% ), url('/img/login.png');
    background-size: cover;
  }
}

@media screen and (max-width:480px) {
  .c3 img {
    max-width: 18rem;
  }
}

@media screen and (max-width:320px) {
  .c3 h1 {
    font-size: 2rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <img
        alt="Logo"
        src="/img/logo.png"
      />
      <h1>
        Seja bem-vindo
      </h1>
    </div>
    <form
      class="c4"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1iich4g-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="email"
          id="email-label"
        >
          Email
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EmailIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
          </div>
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
            id="email"
            name="email"
            placeholder="exemplo@exemplo.com"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1in441m"
            >
              <span>
                Email
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1iich4g-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="password"
          id="password-label"
        >
          Senha
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LockIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
              />
            </svg>
          </div>
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
            id="password"
            name="password"
            placeholder="Digite sua senha"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1in441m"
            >
              <span>
                Senha
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1iich4g-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="confirmPassword"
          id="confirmPassword-label"
        >
          Confirmar senha
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LockIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
              />
            </svg>
          </div>
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
            id="confirmPassword"
            name="confirmPassword"
            placeholder="Confirme sua senha"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1in441m"
            >
              <span>
                Confirmar senha
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-vyxd6w-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Cadastrar
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ecrjkk-MuiTypography-root-MuiLink-root"
      href="/"
    >
      Já possui uma conta? Faça o login!
    </a>
  </div>
</div>
`;
